fala/apps/adviser/views.py

```diff
@@ -37,3 +38,36 @@ def get(self, request, *args, **kwargs):
 
 class AccessibilityView(TemplateView):
     template_name = "adviser/accessibility-statement.html"
+
+
+class SearchView(ListView):
```

Comment:
I can make my peace with this view being unorthodox in that it's very much not single-function and therefore has weirdness like different templates for which thing it's doing. As we discussed, the inability to separate validation from data-fetching means it's not possible to have a normal split between form-handling view (that validates) and results-showing view (that may call the API to do so).

However, this code needs some comments, possibly just in `get_context_data()` but probably in at least `get_template_names()` too, to allow devs to understand the logical flow through this view, because it is non-obvious.

Reply:
I've added some extra comments. In EL-1433 (where there is more logic around the postcodes) we managed to improve this

Reply:
OK great this is good for now and we can pick it up further there